### PR TITLE
[metricbeat] migrate zookeeper to reporter V2 with error handling

### PR DESCRIPTION
**Pull request by fearful-symmetry**
see elastic/beats#11374

Got a little creative here, so we could also clean up the deprecated logging methods. If there's a better way to do that, please tell me.

### CODE REVIEW

File: metricbeat/module/zookeeper/mntr/data.go [hunk 1]
```diff
@@ -72,7 +72,7 @@ func eventMapping(response io.Reader, r mb.ReporterV2) {
 		if match := paramMatcher.FindStringSubmatch(scanner.Text()); len(match) == 3 {
 			fullEvent[match[1]] = match[2]
 		} else {
-			logp.Warn("Unexpected line in mntr output: %s", scanner.Text())
+			logger.Warn("Unexpected line in mntr output: %s", scanner.Text())
```
Comment (ruflin) [inline metricbeat/module/zookeeper/mntr/data.go]: In the past we made the decision that we try not to use `Warn` and always use either Error or Info to make it clear if user action is needed or not. I suggest we change this to Info.

I'm aware this is not introduced by this PR, but as we changed the line stumbled over it. We can change it in this PR or follow up.

File: metricbeat/module/zookeeper/server/data_integration_test.go [hunk 2]
```diff
@@ -1,54 +0,0 @@
-// Licensed to Elasticsearch B.V. under one or more contributor
```
Comment (ruflin) [inline metricbeat/module/zookeeper/server/data_integration_test.go]: +1 on moving this.

File: metricbeat/module/zookeeper/server/server_integration_test.go [hunk 3]
```diff
@@ -58,6 +58,15 @@ func TestFetch(t *testing.T) {
 	}
 }
 
+func TestData(t *testing.T) {
+	//t.Skip("Skipping `data.json` generation test")
```
Comment (ruflin) [inline metricbeat/module/zookeeper/server/server_integration_test.go]: I think we don't need this line?

Comment (fearful-symmetry) [reply metricbeat/module/zookeeper/server/server_integration_test.go]: Yah, I'm...not sure why that was skipped?

**Comment by fearful-symmetry** [comment]
jenkins, test this

**Comment by ruflin** [comment]
I wonder if the test failures are actually related to the change. It says coredns failed but it seems zookeeper is one of the last one to stop. Will run the tests again.

**Comment by ruflin** [comment]
jenkins, test this

**Comment by fearful-symmetry** [comment]
Still seeing this:
`Exception: Timeout while waiting for healthy docker-compose services: coredns`

A bunch of the windows tests seem to be timing out, which I assume is unrelated...

**Comment by ruflin** [comment]
@fearful-symmetry Can you rebase again now that the fix for coredns was merged?

**Pull request merged**